torchaudio/__init__.py

```diff
@@ -338,25 +339,65 @@ def get_sox_bool(i: int = 0) -> Any:
         return _torch_sox.sox_bool(i)
 
 
+_SOX_INITIALIZED = False
+# This variable has a micro lifecycle. (False -> True -> None)
+# False: Not initialized
+# True: Initialized
+# None: Already shut down (should not be initialized again.)
+
+_SOX_SUCCESS_CODE = 0
+# defined at
+# https://fossies.org/dox/sox-14.4.2/sox_8h.html#a8e07e80cebeff3339265d89c387cea93a9ef2b87ec303edfe40751d9a85fadeeb
+
+
 @_audio_backend_guard("sox")
 def initialize_sox() -> int:
-    """Initialize sox for use with effects chains.  This is not required for simple
-    loading.  Importantly, only run `initialize_sox` once and do not shutdown
-    after each effect chain, but rather once you are finished with all effects chains.
-    """
+    """Initialize sox for use with effects chains.
 
-    import _torch_sox
-    return _torch_sox.initialize_sox()
+    You only need to call this function once to use SoX effects chains multiple times.
+    It is safe to call this function multiple times as long as ``shutdown_sox`` is not yet called.
+    Once ``shutdown_sox`` is called, you can no longer use SoX effects and calling this function
+    results in `RuntimeError`.
+
+    Note:
+        This function is not required for simple loading.
+
+    Returns:
+        int: Code corresponding to sox_error_t enum. See
+        https://fossies.org/dox/sox-14.4.2/sox_8h.html#a8e07e80cebeff3339265d89c387cea93
+    """
+    global _SOX_INITIALIZED
+    if _SOX_INITIALIZED is None:
+        raise RuntimeError('SoX effects chain has been already shut down. Can not initialize again.')
```

Comment:
We do not need to do this, but raising `RuntimeError` with a message would be better than segfault.